Update peerDependencies to support Svelte v4

The library is not compatible with Svelte v4. When running `npm install`, we get the following error:
```
npm ERR! Conflicting peer dependency: svelte@3.59.2
npm ERR! node_modules/svelte
npm ERR!   peer svelte@"^3.46.4" from svelte-hamburgers@4.0.2
npm ERR!   node_modules/svelte-hamburgers
npm ERR!     svelte-hamburgers@"^4.0.2" from the root project
```

We need to add Svelte 4 to the peer dependencies.

--- a/package.json
+++ b/package.json
@@ -33,7 +33,7 @@
     "vite": "^4.3.9"
   },
   "peerDependencies": {
-    "svelte": "^3.46.4"
+    "svelte": "^3.46.4 || ^4.0.0"
   },
   "description": "Svelte Hamburgers is a component based on the popular hamburgers.css",
   "author": "GHOST",

